### PR TITLE
Bump version to 3.0.0

### PR DESCRIPTION
**Pull request by nknize**
Bumps main version from 2.0.0 to 3.0.0 after cutting 2.x and 2.0 branches. Some termporary versioning scaffolding is needed for now. This will be cleaned up in a follow up PR.

### CODE REVIEW

**Review by mch2 — APPROVED** [review]

**Comment by nknize** [comment]
Some cleanup is still needed in the version scaffolding

**Comment by nknize** [comment]
start gradle check

File: server/src/main/java/org/opensearch/transport/InboundDecoder.java
Comment (reta) [inline server/src/main/java/org/opensearch/transport/InboundDecoder.java:223]: Should we introduce `NEXT` constant for that? ( `== Version.fromId(4000000)` for now)

**Review by reta — APPROVED** [review]

File: server/src/main/java/org/opensearch/transport/InboundDecoder.java
Comment (nknize) [reply server/src/main/java/org/opensearch/transport/InboundDecoder.java:222]: :+1:

**Comment by tlfeng** [comment]
In log 3988:
```
REPRODUCE WITH: ./gradlew ':distribution:tools:plugin-cli:test' --tests "org.opensearch.plugins.RemovePluginCommandTests.testRemoveOldVersion" -Dtests.seed=63141A04D4E4F2B3 -Dtests.security.manager=false -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es-EC -Dtests.timezone=Pacific/Marquesas -Druntime.java=17

org.opensearch.plugins.RemovePluginCommandTests > testRemoveOldVersion FAILED
    java.lang.IllegalArgumentException: maxVersion [1.3.1] cannot be less than minVersion [2.0.0]
        at __randomizedtesting.SeedInfo.seed([63141A04D4E4F2B3:6868F792A67F961E]:0)
        at org.opensearch.test.VersionUtils.randomVersionBetween(VersionUtils.java:324)
        at org.opensearch.plugins.RemovePluginCommandTests.testRemoveOldVersion(RemovePluginCommandTests.java:173)
```
```
REPRODUCE WITH: ./gradlew ':test:framework:test' --tests "org.opensearch.test.rest.yaml.section.SkipSectionTests.testSkipMultiRange" -Dtests.seed=63141A04D4E4F2B3 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=lt -Dtests.timezone=America/Guyana -Druntime.java=17

org.opensearch.test.rest.yaml.section.SkipSectionTests > testSkipMultiRange FAILED
    java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([63141A04D4E4F2B3:79A142FD42A589AB]:0)
        at org.junit.Assert.fail(Assert.java:87)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.junit.Assert.assertTrue(Assert.java:53)
        at org.opensearch.test.rest.yaml.section.SkipSectionTests.testSkipMultiRange(SkipSectionTests.java:64)
```

**Comment by tlfeng** [comment]
In log 4015
```
* What went wrong:
Execution failed for task ':qa:mixed-cluster:v2.1.0#mixedClusterTest'.
> `cluster{:qa:mixed-cluster:v2.1.0}` failed to wait for cluster health yellow after 40 SECONDS
    IO error while waiting cluster
    503 Service Unavailable
```

start gradle check

**Comment by tlfeng** [comment]
In log 4017
the same transient issue
```
* What went wrong:
Execution failed for task ':qa:mixed-cluster:v2.1.0#mixedClusterTest'.
> `cluster{:qa:mixed-cluster:v2.1.0}` failed to wait for cluster health yellow after 40 SECONDS
    IO error while waiting cluster
    503 Service Unavailable
```

**Comment by tlfeng** [comment]
start gradle check

**Comment by tlfeng** [comment]
start gradle check

**Comment by tlfeng** [comment]
In log 4019:
```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.cluster.coordination.JoinTaskExecutorTests.testPreventJoinClusterWithUnsupportedNodeVersions" -Dtests.seed=E91B43DAF672B0E7 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=it-IT -Dtests.timezone=America/Barbados -Druntime.java=17

org.opensearch.cluster.coordination.JoinTaskExecutorTests > testPreventJoinClusterWithUnsupportedNodeVersions FAILED
    junit.framework.AssertionFailedError: Expected exception IllegalStateException but no exception was thrown
        at __randomizedtesting.SeedInfo.seed([E91B43DAF672B0E7:6A64E4A024B120AB]:0)
        at org.apache.lucene.tests.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2898)
        at org.apache.lucene.tests.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2884)
        at org.opensearch.cluster.coordination.JoinTaskExecutorTests.testPreventJoinClusterWithUnsupportedNodeVersions(JoinTaskExecutorTests.java:119)
```

**Comment by nknize** [comment]
This is blocked until #2722 is merged and backported to 2.x and 2.0 branches.

See failure:

```
» WARN ][o.o.c.c.JoinHelper       ] [v2.1.0-0] last failed join attempt was 749ms ago, failed to join {v2.1.0-2}{tF0gn1KRTXiDyL3QaiZd3w}{3m_3VtzOTjung38rji5Hew}{127.0.0.1}{127.0.0.1:39911}{dimr}{testattr=test, shard_indexing_pressure_enabled=true} with JoinRequest{sourceNode={v2.1.0-0}{JOUu8X0cSU6fIIPyBpsSqA}{tERcNWxcQqeaPYPs2cpynQ}{127.0.0.1}{127.0.0.1:33229}{dimr}{upgraded=true, testattr=test, shard_indexing_pressure_enabled=true}, minimumTerm=6, optionalJoin=Optional.empty}
»  org.opensearch.transport.RemoteTransportException: [v2.1.0-2][127.0.0.1:39911][internal:cluster/coordination/join]
»  Caused by: org.opensearch.transport.ConnectTransportException: [v2.1.0-0][127.0.0.1:33229] general node connection failure
..
..
..
»  Caused by: java.lang.IllegalStateException: Received message from unsupported version: [3.0.0] minimal compatible version is: [7.10.0]
```

#2722 will stage 2.0+ clusters to only spoof major versions when running in mixed mode w/ legacy 7.x nodes.

**Pull request merged**
